File: iOS Car Dealer Tracker/Ezcar24Business/Models/CRMModels.swift
import Foundation

struct FunnelMetrics {
    let countsPerStage: [LeadStage: Int]
    let conversionRates: [LeadStage: Double]
    let totalLeads: Int
    let wonLeads: Int
    let lostLeads: Int
    let activeLeads: Int
    let overallConversionRate: Double
}

struct SourcePerformance {
    let source: LeadSource
    let leadCount: Int
    let convertedCount: Int
    let conversionRate: Double
    let totalValue: Decimal
}

struct PipelineSummary {
    let totalPipelineValue: Decimal
    let weightedPipelineValue: Decimal
    let averageDealSize: Decimal
    let expectedRevenue: Decimal
}

struct DailyActivitySummary {
    let callsCount: Int
    let meetingsCount: Int
    let messagesCount: Int
    let newLeadsCount: Int
    let followUpsCount: Int
    let totalInteractions: Int
    let date: Date
}

struct LeadScoreBreakdown {
    let totalScore: Int
    let interactionScore: Int
    let recencyScore: Int
    let engagementScore: Int
    let valueScore: Int
}

struct StageTransition {
    let fromStage: LeadStage
    let toStage: LeadStage
    let daysInStage: Int
    let transitionedAt: Date?
}

struct LeadSummary {
    let clientId: UUID
    let clientName: String
    let currentStage: LeadStage
    let estimatedValue: Decimal?
    let probability: Double
    let daysInCurrentStage: Int
    let lastContactAt: Date?
    let nextFollowUpAt: Date?
}

File: Android Car Dealer Tracker/app/src/main/java/com/ezcar24/business/util/calculator/CRMModels.kt
package com.ezcar24.business.util.calculator

import com.ezcar24.business.data.local.LeadSource
import com.ezcar24.business.data.local.LeadStage
import java.math.BigDecimal

/**
 * Data class representing funnel metrics for lead tracking
 */
data class FunnelMetrics(
    val countsPerStage: Map<LeadStage, Int>,
    val conversionRates: Map<Pair<LeadStage, LeadStage>, Double>,
    val averageTimePerStage: Map<LeadStage, Int>,
    val totalLeads: Int,
    val wonLeads: Int,
    val lostLeads: Int,
    val activeLeads: Int,
    val overallConversionRate: Double
)

/**
 * Data class representing daily activity summary for CRM
 */
data class DailyActivitySummary(
    val callsCount: Int,
    val meetingsCount: Int,
    val messagesCount: Int,
    val newLeadsCount: Int,
    val followUpsCount: Int,
    val totalInteractions: Int,
    val date: java.util.Date
)

/**
 * Data class representing lead source performance metrics
 */
data class SourcePerformance(
    val source: LeadSource,
    val leadCount: Int,
    val convertedCount: Int,
    val conversionRate: Double,
    val totalValue: BigDecimal,
    val averageLeadValue: BigDecimal
)

/**
 * Data class representing lead score breakdown
 */
data class LeadScoreBreakdown(
    val totalScore: Int,
    val interactionScore: Int,
    val recencyScore: Int,
    val engagementScore: Int,
    val valueScore: Int
)

/**
 * Data class representing stage transition info
 */
data class StageTransition(
    val fromStage: LeadStage,
    val toStage: LeadStage,
    val daysInStage: Int,
    val transitionedAt: java.util.Date?
)

/**
 * Data class representing pipeline summary
 */
data class PipelineSummary(
    val totalPipelineValue: BigDecimal,
    val weightedPipelineValue: BigDecimal,
    val leadsByStage: Map<LeadStage, List<LeadSummary>>,
    val averageDealSize: BigDecimal,
    val expectedRevenue: BigDecimal
)

/**
 * Data class representing a lead summary for pipeline view
 */
data class LeadSummary(
    val clientId: java.util.UUID,
    val clientName: String,
    val currentStage: LeadStage,
    val estimatedValue: BigDecimal?,
    val probability: Double,
    val daysInCurrentStage: Int,
    val lastContactAt: java.util.Date?,
    val nextFollowUpAt: java.util.Date?
)
